src/main/webapp/query-control.js

```diff
@@ -35,13 +35,82 @@ function clearPreviousResult() {
   document.getElementById('result').style.display = 'block';
 }
 
-function getFetchUrl(personType, action, location, year) {
-  return '/query?person-type=' + personType +
+function getFetchUrl(servlet, personType, action, location, year) {
+  return '/' + servlet + '?person-type=' + personType +
     '&action=' + action +
     '&location=' + location +
     '&year=' + year;
 }
 
+function getTitle(personType, location, year, locationInput, actionInput) {
```

Comment:
This is a nice modularization. There's a spot later in the code that could use it as well (currently the function is basically just copied in two places).

Reply:
get HistoryUrl now calls to getFetchUrl to eliminate repeated code